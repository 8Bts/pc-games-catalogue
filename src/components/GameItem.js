import PropTypes from 'prop-types';
import item from '../styles/item.module.css';

const GameItem = ({
  name, relDate, genres, img,
}) => {
  const background = {
    backgroundImage: `url(${img})`,
  };

  return (
    <li className="list-group-item">
      <div className="row">
        <div className={`${item.left} col-4`}>
          <span className={item.img} style={background} />
        </div>
        <div className={`${item.right} col-8`}>
          <h3>{name}</h3>
          <div className="mb-3">
            {genres.map((genre) => (<span className="badge bg-info me-2" key={genre.id}>{genre.name}</span>))}
          </div>
          <span>
            Released:
            {`  ${relDate}`}
          </span>
        </div>
      </div>
    </li>
  );
};

GameItem.propTypes = {
  name: PropTypes.string,
  relDate: PropTypes.string,
  genres: PropTypes.arrayOf(PropTypes.object),
  img: PropTypes.string,
};

GameItem.defaultProps = {
  name: '',
  relDate: '',
  genres: [],
  img: '',
};

export default GameItem;
