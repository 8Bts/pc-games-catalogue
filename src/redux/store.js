import { createStore, applyMiddleware } from 'redux';
import promise from 'redux-promise-middleware';
import reducer from './reducers/index';

export default createStore(reducer, {}, applyMiddleware(promise));
