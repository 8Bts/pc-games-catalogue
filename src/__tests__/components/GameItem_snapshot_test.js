import toJson from 'enzyme-to-json';
import { mount, configure } from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import GameItem from '../../components/GameItem';
import store from '../../redux/store';

configure({ adapter: new Adapter() });
it('Renders correctly', () => {
  const wrapper = mount(
    <Provider store={store}>
      <BrowserRouter>
        <GameItem
          id={1}
          name="Grand Theft Auto V"
          relDate="01/01/11"
          genres={[
            {
              id: 1,
              name: 'action',
            },
            {
              id: 2,
              name: 'indie',
            },
          ]}
          img=""
          rating={4}
        />
      </BrowserRouter>
    </Provider>,
  );
  expect(toJson(wrapper)).toMatchSnapshot();
});
