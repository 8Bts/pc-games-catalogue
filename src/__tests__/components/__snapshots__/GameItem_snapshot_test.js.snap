// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders correctly 1`] = `
<Provider
  store={
    Object {
      "@@observable": [Function],
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <BrowserRouter>
    <Router
      history={
        Object {
          "action": "POP",
          "block": [Function],
          "createHref": [Function],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
          "push": [Function],
          "replace": [Function],
        }
      }
    >
      <GameItem
        genres={
          Array [
            Object {
              "id": 1,
              "name": "action",
            },
            Object {
              "id": 2,
              "name": "indie",
            },
          ]
        }
        id={1}
        img=""
        name="Grand Theft Auto V"
        rating={4}
        relDate="01/01/11"
      >
        <li
          className="list-group-item li"
        >
          <div
            className="row"
          >
            <div
              className="left col-md-4"
            >
              <Link
                to="/game/1"
              >
                <LinkAnchor
                  href="/game/1"
                  navigate={[Function]}
                >
                  <a
                    href="/game/1"
                    onClick={[Function]}
                  >
                    <span
                      className="img"
                      style={
                        Object {
                          "backgroundImage": "url()",
                        }
                      }
                    />
                  </a>
                </LinkAnchor>
              </Link>
            </div>
            <div
              className="right col-md-8"
            >
              <h3>
                <Link
                  className="anchor"
                  to="/game/1"
                >
                  <LinkAnchor
                    className="anchor"
                    href="/game/1"
                    navigate={[Function]}
                  >
                    <a
                      className="anchor"
                      href="/game/1"
                      onClick={[Function]}
                    >
                      Grand Theft Auto V
                    </a>
                  </LinkAnchor>
                </Link>
              </h3>
              <div
                className="mb-3"
              >
                <span
                  className="badge bg-info me-2"
                  key="1"
                >
                  action
                </span>
                <span
                  className="badge bg-info me-2"
                  key="2"
                >
                  indie
                </span>
                <span
                  className="badge bg-warning"
                >
                  4
                   / 5
                </span>
              </div>
              <span>
                Released:
                  01/01/11
              </span>
            </div>
          </div>
        </li>
      </GameItem>
    </Router>
  </BrowserRouter>
</Provider>
`;
